### PR TITLE
Implement ZM's token auth API

### PR DESCRIPTION
**Pull request by mnoorenberghe**
Tested on ZM's master branch and with version 1.32.3

### CODE REVIEW

File: zoneminder/zm.py [hunk 1]
```diff
@@ -72,12 +92,17 @@ def change_state(self, api_url, post_data) -> dict:
...
+        if self._auth_token:
+            token_url_suffix = "?token=" + self._auth_token
         try:
             # Since the API uses sessions that expire, sometimes we need to
             # re-auth if the call fails.
             for _ in range(ZoneMinder.LOGIN_RETRIES):
                 req = requests.request(
-                    method, urljoin(self._server_url, api_url), data=data,
+                    method,
+                    urljoin(self._server_url, api_url) + token_url_suffix,
```
Comment (rohankapoorcom) [inline zoneminder/zm.py]: Can we do this with a `urljoin` rather than string concat?

Comment (mnoorenberghe) [reply zoneminder/zm.py]: `urljoin` can't join query parameters, the second argument would just override the arguments passed as the base. What problem are you trying to solve?

Comment (mnoorenberghe) [reply zoneminder/zm.py]: @rohankapoorcom ^

Comment (rohankapoorcom) [reply zoneminder/zm.py]: Sorry - I was coming at this from a Java standpoint.... I was trying to avoid the string concats.

**Comment by rohankapoorcom** [comment]
Released in v0.4.0. Thanks @mnoorenberghe

**Pull request merged**